go/parquet/pqarrow/schema.go

```diff
@@ -406,6 +406,11 @@ func fieldToNode(name string, field arrow.Field, props *parquet.WriterProperties
 		return nil, fmt.Errorf("%w: support for %s", arrow.ErrNotImplemented, field.Type.ID())
 	}
 
+	if customLogicalType != nil {
+		logicalType = customLogicalType.Type
+		length = customLogicalType.Length
```

Comment:
The `length` here should come from the physical layout. The `LogicalType` interface has method `IsApplicable(t parquet.Type, tlen int32) bool` that can validate whether the underlying type is of the right size for the logical type.

In practice this means you shouldn't need to specify the length here at all. The arrow types above that correspond to `FixedLenByteArray` should already be setting `length` to the appropriate value.

That raises another point. If `schema.LogicalType` is actually the only field that the new `LogicalType` struct needs to track, then we can skip defining the new struct type and just use the `schema.LogicalType` interface instead.

Reply:
Thanks for explaining.
I have updated to use `schema.LogicalType` directly 3de0781